components/wifi/esp8266-driver/ESP8266Interface.h

```diff
@@ -30,7 +30,9 @@
 #include "features/netsocket/WiFiAccessPoint.h"
 #include "features/netsocket/WiFiInterface.h"
 #include "platform/Callback.h"
+#if MBED_CONF_RTOS_PRESENT
```

Comment:
Why is this needed? same MACRO guardian is already in that header file. And if the header file does not compile as such (includes rtos headers outside MACRO) those headers should be moved inside MBED_CONF_RTOS_PRESENT macro.